client/utils/random.c

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) 2009-2010 Satoshi Nakamoto
+// Copyright (c) 2009-2016 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include "random.h"
+#ifdef _WIN32
+#include <windows.h>
+#else
+#include <sys/time.h>
+#endif
+#include <openssl/rand.h>
+#include "log.h"
+
+static void RandFailure()
+{
+	xdag_fatal("Failed to read randomness, aborting\n");
+	abort();
+}
+
+static inline int64_t GetPerformanceCounter()
+{
+	int64_t counter = 0;
+#ifdef _WIN32
+	QueryPerformanceCounter((LARGE_INTEGER*)&counter);
+#else
+	struct timeval t;
+	gettimeofday(&t, NULL);
+	counter = (int64_t)(t.tv_sec * 1000000 + t.tv_usec);
+#endif
+	return counter;
+}
+
+void RandAddSeed()
```

Comment:
If i understood correctly, in unix systems the seed is took from /dev/random and windows with RAND_screen();
After this, to add more entropy to the seed, it add the time. Is it right ?

Reply:
I'm not sure about /dev/random, but yes, time stamp is used for seeding the OpenSSL random generator. It is also possible to use some screen data for seeding on Windows OS.